pkg/operator/staticpod/controller/node/node_controller_test.go

```diff
@@ -26,6 +26,129 @@ func fakeMasterNode(name string) *v1.Node {
 	return n
 }
 
+func makeNodeNotReady(node *v1.Node) *v1.Node {
+	con := v1.NodeCondition{}
+	con.Type = v1.NodeReady
+	con.Status = v1.ConditionFalse
+	node.Status.Conditions = append(node.Status.Conditions, con)
+	return node
+}
+
+func TestNodeControllerDegradedConditionType(t *testing.T) {
+	scenarios := []struct {
+		name               string
+		masterNodes        []runtime.Object
+		evaluateNodeStatus func([]operatorv1.OperatorCondition) error
+	}{
+		// scenario 1
+		{
+			name:        "scenario 1: one unhealthy master node is reported",
+			masterNodes: []runtime.Object{makeNodeNotReady(fakeMasterNode("test-node-1")), fakeMasterNode("test-node-2")},
+			evaluateNodeStatus: func(conditions []operatorv1.OperatorCondition) error {
+				if len(conditions) != 1 {
+					return fmt.Errorf("expected exaclty 1 condition, got %d", len(conditions))
+				}
+
+				condition := conditions[0]
+				if condition.Type != "NodeControllerDegraded" {
```

Comment:
why not use `NodeControllerDegradedConditionType` ?

Reply:
because I don't like to use "production code" in the tests :) Anyway I'll change that :)